crates/red_knot_python_semantic/src/site_packages.rs

```diff
@@ -370,18 +376,24 @@ fn site_packages_directory_from_sys_prefix(
 ///
 /// [`sys.prefix`]: https://docs.python.org/3/library/sys.html#sys.prefix
 #[derive(Debug, PartialEq, Eq, Clone)]
-pub(crate) struct SysPrefixPath(SystemPathBuf);
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct SysPrefixPath {
+    pub(crate) inner: SystemPathBuf,
+    pub(crate) origin: SysPrefixPathOrigin,
+}
 
 impl SysPrefixPath {
```

Comment:
it might be slightly better to keep these fields private and have `pub(crate)` accessor methods:

```suggestion
pub struct SysPrefixPath {
    inner: SystemPathBuf,
    origin: SysPrefixPathOrigin,
}

impl SysPrefixPath {
    pub(crate) fn inner(&self) -> &SystemPath {
        &self.inner
    }
    
    pub(crate) fn origin(&self) -> SysPrefixPathOrigin {
        self.origin
    }

```

though it probably doesn't matter that much in practice!